src/dc_msg.rs

```diff
@@ -831,98 +832,74 @@ pub unsafe fn dc_msg_get_summarytext(
         return dc_strdup(0 as *const libc::c_char);
     }
 
-    let msgtext_c = (*msg)
-        .text
-        .as_ref()
-        .map(|s| CString::yolo(String::as_str(s)));
-    let msgtext_ptr = msgtext_c.map_or(ptr::null(), |s| s.as_ptr());
-
     dc_msg_get_summarytext_by_raw(
         (*msg).type_0,
-        msgtext_ptr,
+        (*msg).text.as_ref().unwrap(),
         &mut (*msg).param,
         approx_characters,
         (*msg).context,
     )
+    .strdup()
 }
 
-/* the returned value must be free()'d */
+/// get a summary text
 #[allow(non_snake_case)]
-pub unsafe fn dc_msg_get_summarytext_by_raw(
+pub fn dc_msg_get_summarytext_by_raw(
     type_0: Viewtype,
-    text: *const libc::c_char,
+    text: &str,
     param: &mut Params,
     approx_characters: libc::c_int,
     context: &Context,
-) -> *mut libc::c_char {
-    /* get a summary text, result must be free()'d, never returns NULL. */
-    let mut ret;
-    let mut prefix: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut pathNfilename: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut value: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut append_text: libc::c_int = 1i32;
+) -> String {
+    let ret: String;
+    let mut prefix = "".to_string();
+    let mut append_text = true;
     match type_0 {
-        Viewtype::Image => prefix = context.stock_str(StockMessage::Image).strdup(),
-        Viewtype::Gif => prefix = context.stock_str(StockMessage::Gif).strdup(),
-        Viewtype::Video => prefix = context.stock_str(StockMessage::Video).strdup(),
-        Viewtype::Voice => prefix = context.stock_str(StockMessage::VoiceMessage).strdup(),
+        Viewtype::Image => prefix = context.stock_str(StockMessage::Image).to_string(),
```

Comment:
why `to_string()` isn't it already returning a string?

Reply:
nvm, it's something about str vs. String...

Reply:
And it's also inside a `Cow<T>`

Reply:
AFAIK you could also make this more explicit with `.to_owned()` which is what I've been doing in these cases.  It's kind of unfortunate that we have to copy though (insert dig about dynamic languages not having issues like this).